tests/integration/test_spanner_graph_qa.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+import string
+
+import pytest
+from google.cloud import spanner
+from langchain.evaluation import load_evaluator
+from langchain_community.graphs.graph_document import GraphDocument, Node, Relationship
+from langchain_core.documents import Document
+from langchain_google_vertexai import ChatVertexAI, VertexAIEmbeddings
+
+from langchain_google_spanner.graph_qa import SpannerGraphQAChain
+from langchain_google_spanner.graph_store import SpannerGraphStore
+
+project_id = os.environ["PROJECT_ID"]
+instance_id = os.environ["INSTANCE_ID"]
+database_id = os.environ["GOOGLE_DATABASE"]
+
+
+def random_string(num_char=3, exclude_whitespaces=False):
```

Comment:
it seems  `exclude_whitespaces` is always True in your tests, so maybe simplify the function a bit by removing it?

Reply:
done